src/projects/project.vala

```diff
@@ -126,7 +126,7 @@ abstract class Vls.Project : Object {
         }
         for (int i = last_idx - 1; i >= 0; i--) {
```

Comment:
I think you may also need to make a similar change to this outer loop.

Reply:
Actually, you don't. Because `build_targets[last_idx]` must be a `Compilation`, or it should be removed, and this check already happens.